templates/tomcat/tomcat.conf.erb

```diff
@@ -0,0 +1,55 @@
+# This is managed by Puppet, DO NOT EDIT!
+# System-wide configuration file for tomcat services
+# This will be loaded by systemd as an environment file,
+# so please keep the syntax. For shell expansion support
+# place your custom files as /etc/tomcat/conf.d/*.conf
+#
+# There are 2 "classes" of startup behavior in this package.
+# The old one, the default service named tomcat.service.
+# The new named instances are called tomcat@instance.service.
+#
+# Use this file to change default values for all services.
+# Change the service specific ones to affect only one service.
+# For tomcat.service it's /etc/sysconfig/tomcat, for
+# tomcat@instance it's /etc/sysconfig/tomcat@instance.
+
+# This variable is used to figure out if config is loaded or not.
+TOMCAT_CFG_LOADED="<%= scope['::candlepin::tomcat_cfg_loaded'] %>"
+
+# In new-style instances, if CATALINA_BASE isn't specified, it will
+# be constructed by joining TOMCATS_BASE and NAME.
+TOMCATS_BASE="<%= scope['::candlepin::tomcat_base'] %>"
+
+# Where your java installation lives
+JAVA_HOME="<%= scope['::candlepin::java_home'] %>"
+
+# Where your tomcat installation lives
+CATALINA_HOME="<%= scope['::candlepin::catalina_home'] %>"
+
+# System-wide tmp
+CATALINA_TMPDIR="<%= scope['::candlepin::catalina_tmpdir'] %>"
+
+# You can pass some parameters to java here if you wish to
+JAVA_OPTS="<%= scope['::candlepin::java_opts'] %>"
+
+# You can change your tomcat locale here
+<% unless [nil, :undefined, :undef, '', 0].include?(scope.lookupvar('::candlepin::lang')) %>
+LANG="<%= scope['::candlepin::lang'] %>"
+<% else %>
+#LANG="en_US"
```

Comment:
What's the benefit of this commented out parameter? Since puppet controls this, no one can enable this except through the installer.

Reply:
I feel like we should state that then. What I see our users doing is uncommenting this, restarting tomcat, upgrading and then being mad because we wiped out their configuration.

Reply:
The very first line states it's mentioned by puppet.

Reply:
Fair enough. But knowing our users, that header gets overlooked.

Reply:
Oh sure, but we can at least tell them we warned them in the header ;)